packages/teleterm/src/ui/TopBar/Identity/useIdentity.ts

```diff
@@ -15,20 +15,8 @@ export function useIdentity() {
     ctx.commandLauncher.executeCommand('cluster-connect', {});
   }
 
-  function removeCluster(clusterUri: string): void {
-    ctx.commandLauncher.executeCommand('cluster-remove', { clusterUri });
-  }
-
-  async function logout(): Promise<void> {
-    const rootClusterUri = ctx.workspacesService.getRootClusterUri();
-    await ctx.clustersService.logout(rootClusterUri);
-    const [firstConnectedWorkspace] =
-      ctx.workspacesService.getConnectedWorkspacesClustersUri();
-    if (firstConnectedWorkspace) {
-      await ctx.workspacesService.setActiveWorkspace(firstConnectedWorkspace);
-    } else {
-      await ctx.workspacesService.setActiveWorkspace(null);
-    }
+  async function logout(clusterUri: string): Promise<void> {
+    ctx.commandLauncher.executeCommand('cluster-logout', {clusterUri})
```

Comment:
We have to be on the lookout when it comes to `ctx.commandLauncher`, I have a feeling that it's becoming a place to dump any business logic that doesn't really fit anywhere else. 😅

Maybe we'll get some better ideas in the future.

Reply:
Maybe we should switch to more specialised services for these actions? (in the future)

Reply:
Yeah, idk, it looks like most services in our code are those long-lived objects that are stored in the app context.

But if you look at the commands in the command launcher, they need to just orchestrate the work of a single or a couple of services (with the exception of `tsh-ssh` which definitely could be refactored to keep all of that logic elsewhere 🙊). The way I think about it is that those commands play a similar role as the controller layer in MVC.

I think having a layer like that is useful, maybe it just shouldn't be tied to the concept of a command launcher (since we don't really have a command launcher) and instead of operating on strings to identify the commands it could just export a couple of functions.